lib/private/machines/create-manager.js

```diff
@@ -114,25 +114,33 @@ module.exports = {
       normalizeDatastoreConfig(_clientConfig, CONFIG_WHITELIST, EXPECTED_URL_PROTOCOL_PFX);
     } catch (e) {
       switch (e.code) {
-        case 'E_BAD_CONFIG': return exits.malformed({ error: e, meta: undefined });
-        default: return exits.error(e);
+        case 'E_BAD_CONFIG':
+          return exits.malformed({error: e, meta: undefined});
+        default:
+          return exits.error(e);
       }
     }
 
     // Mongo doesn't like some of our standard properties, so we'll remove them
     // (we don't need any of them now anyway, since we know at this point that
     // they'll have been baked into the URL)
     var mongoUrl = _clientConfig.url;
+    var database = _clientConfig.database;
     _clientConfig = _.omit(_clientConfig, ['url', 'user', 'password', 'host', 'port', 'database']);
 
-    NodeMongoDBNativeLib.MongoClient.connect(mongoUrl, _clientConfig, function connectCb(err, db) {
+    // so that it uses the new URL string parser. Old one is deprecated
+    _.merge(_clientConfig, {useNewUrlParser: true});
```

Comment:
Could we get away with a shallow Object.assign() aka _.extend() instead of a deep merge?